source/extensions/filters/common/lua/lua.cc

```diff
@@ -71,8 +71,9 @@ int ThreadLocalState::getGlobalRef(uint64_t slot) {
 }
 
 uint64_t ThreadLocalState::registerGlobal(const std::string& global) {
-  tls_slot_->runOnAllThreads([this, global]() {
-    LuaThreadLocal& tls = tls_slot_->getTyped<LuaThreadLocal>();
+  tls_slot_->runOnAllThreads([global](ThreadLocal::ThreadLocalObjectSharedPtr ptr) -> auto {
```

Comment:
Does spelling out `-> auto` (`-> ThreadLocal::ThreadLocalObjectSharedPtr`) here will be a problem for clang-tidy?

Reply:
And let's rename `ptr` as `previous`?

Reply:
From the existing code, it is acceptable to spell out the return type or not.

Reply:
Anyway, I will remove it to make the code more concise.